android/src/main/java/co/nimblehq/ic/kmm/suv/android/ui/screens/home/HomeViewModel.kt

```diff
@@ -24,24 +26,63 @@ class HomeViewModel(
     private val _avatarUrlString = MutableStateFlow("")
     val avatarUrlString: StateFlow<String> = _avatarUrlString.asStateFlow()
 
+    private val _surveysUiModel: MutableStateFlow<HomeSurveysUiModel?> = MutableStateFlow(null)
+    val surveysUiModel: StateFlow<HomeSurveysUiModel?> = _surveysUiModel.asStateFlow()
+
+    private lateinit var surveys: List<Survey>
+    private var currentSurveyIndex: Int = 0
+
     init {
         _currentDate.value = dateTimeFormatter.getFormattedString(
             dateTime.today(),
             DateFormat.WeekDayMonthDay
         ).uppercase()
     }
 
-    fun loadProfile() {
+    fun loadProfileAndSurveys() {
         showLoading()
         viewModelScope.launch {
-            getProfileUseCase()
+            val getProfile = getProfileUseCase()
+            // TODO: Improve this when we have paging implementation
+            val getSurveys = getSurveysUseCase(1, 5)
+            getProfile
+                .combine(getSurveys) { profile, surveys -> Pair(profile, surveys) }
                 .catch { e ->
                     showError(e.message)
                 }
                 .collect {
-                    _avatarUrlString.value = it.avatarUrl
+                    _avatarUrlString.value = it.first.avatarUrl
+                    surveys = it.second
+                    _surveysUiModel.value = HomeSurveysUiModel(
+                        surveys = surveys.map { survey ->
+                            HomeSurveyUiModel(
+                                survey.title,
+                                survey.description,
+                                survey.coverImageUrl
+                            )
+                        },
+                        currentPageIndex = 0
+                    )
                 }
             hideLoading()
         }
     }
+
+    fun showPreviousSurvey() {
+        if (currentSurveyIndex > 0) {
+            currentSurveyIndex -= 1
+            updateCurrentSurvey()
+        }
+    }
+
+    fun showNextSurvey() {
+        if (currentSurveyIndex < surveys.size - 1) {
```

Comment:
Can we use _surveysUiModel?

```suggestion
        if (currentSurveyIndex < _surveysUiModel.value.second.size - 1) {
```

Reply:
Replied [here](https://github.com/suho/kmm-ic/pull/107#discussion_r1039454456). What do you think?